rskj-core/src/test/java/co/rsk/peg/federation/PendingFederationTest.java

```diff
@@ -137,96 +137,153 @@ void testToString() {
-        testBuildFederation(
-            true,
-            true,
-            false,
-            NetworkParameters.ID_TESTNET,
-            6
+    @ParameterizedTest
+    @MethodSource("networkParameters")
+    void buildFederation_with6Members_afterRSKIP201Activation_beforeRSKIP284Activation_shouldBuildNonStandardERPFed(NetworkParameters networkParams, FederationConstants federationConstants) {
+        // Arrange
+        ActivationConfig.ForBlock activations = ActivationConfigsForTest.iris300().forBlock(FEDERATION_CREATION_BLOCK_NUMBER);
+        FederationArgs federationArgs = new FederationArgs(federationMembers, creationTime, FEDERATION_CREATION_BLOCK_NUMBER, networkParams);
+        List<BtcECKey> erpPubKeys = federationConstants.getErpFedPubKeysList();
+        long activationDelay = federationConstants.getErpFedActivationDelay();
+        Federation expectedFederation = FederationFactory.buildNonStandardErpFederation(federationArgs, erpPubKeys, activationDelay, activations);
+
+        // Act
+        PendingFederation otherPendingFederation = new PendingFederation(
+            federationMembers
         );
+        Federation builtFederation = otherPendingFederation.buildFederation(
+            creationTime,
+            FEDERATION_CREATION_BLOCK_NUMBER,
+            federationConstants,
+            activations
+        );
+
+        // Assert
+        assertEquals(expectedFederation, builtFederation);
     }
 
-    @Test
-    void buildFederation_erp_ok_after_RSKIP_201_activation_after_RSKIP284_mainnet() {
-        testBuildFederation(
-            true,
-            true,
-            false,
-            NetworkParameters.ID_MAINNET,
-            6
+    @ParameterizedTest
+    @MethodSource("networkParameters")
+    void buildFederation_with6Members_afterRSKIP284Activations_shouldBuildNonStandardERPFed(NetworkParameters networkParams, FederationConstants federationConstants) {
+        // Arrange
+        ActivationConfig.ForBlock activations = ActivationConfigsForTest.hop400().forBlock(FEDERATION_CREATION_BLOCK_NUMBER);
+        FederationArgs federationArgs = new FederationArgs(federationMembers, creationTime, FEDERATION_CREATION_BLOCK_NUMBER, networkParams);
+        List<BtcECKey> erpPubKeys = federationConstants.getErpFedPubKeysList();
+        long activationDelay = federationConstants.getErpFedActivationDelay();
+        Federation expectedFederation = FederationFactory.buildNonStandardErpFederation(federationArgs, erpPubKeys, activationDelay, activations);
+
+        // Act
+        PendingFederation otherPendingFederation = new PendingFederation(
+            federationMembers
+        );
+        Federation builtFederation = otherPendingFederation.buildFederation(
+            creationTime,
+            FEDERATION_CREATION_BLOCK_NUMBER,
+            federationConstants,
+            activations
         );
+
+        // Assert
+        assertEquals(expectedFederation, builtFederation);
```

Comment:
The assertion of this test case is the same as the test case preRSKIP284. RSKIP284 is related to the NonStandardErpRedeemScriptBuilderHardcoded. What if we add one more assertion checking that after RSKIP284 the resulted script is different than the hardcoded?

Reply:
Not sure If I understood correctly, there is this test `buildFederation_with6Members_afterRSKIP201Activation_beforeRSKIP284Activation_inTestnet_shouldHaveANonStandardERPRedeemScript` that asserts the following:

`assertEquals(NON_STANDARD_ERP_REDEEM_SCRIPT_HARDCODED, builtFederation.getRedeemScript());`

Is that what you meant?

Reply:
Exactly, we should assert that after RSKIP284, the built federation does not have the NON_STANDARD_ERP_REDEEM_SCRIPT_HARDCODED script.

Reply:
That's already tested https://github.com/rsksmart/rskj/blob/11b31711341cc102f2ab8562fa236be99d6d158b/rskj-core/src/test/java/co/rsk/peg/federation/PendingFederationTest.java#L183